src/rdpc/analysesProcessor.ts

```diff
@@ -0,0 +1,281 @@
+      },
+      page: {from: ${from}, size: ${size}}
+    ) {
+      ...AnalysisData
+      runs: inputForRuns {
+        runId
+        state
+        sessionId
+        repository
+        inputAnalyses {
+          analysisId
+        }
+      }
+    }
+  }
+  `;
+
+  return query;
+};
+
+const retryConfig = {
+  factor: 2,
+  retries: 5,
+  minTimeout: 1000,
+  maxTimeout: Infinity,
+};
+
+export const fetchSeqExpAnalyses = async (
+  studyId: string,
+  url: string,
+  from: number,
+  size: number
+): Promise<Analysis[]> => {
+  const query = buildQuery(studyId, from, size);
+
+  return await promiseRetry<Analysis[]>(async (retry) => {
+    try {
+      logger.info("Fetching sequencing experiment analyses from rdpc.....");
+      const response = await fetch(url, {
+        method: "POST",
+        body: JSON.stringify({ query }),
+        headers: {
+          "Content-type": "application/json",
+        },
+      });
+      return (await response.json()).data
+        .SequencingExperimentAnalyses as Analysis[];
+    } catch (err) {
+      logger.warn(
+        `Failed to fetch sequencing experiment analyses: ${err}, retrying...`
+      );
+      return retry(err);
+    }
+  }, retryConfig).catch((err) => {
+    logger.error(
+      `Failed to fetch sequencing experiment analyses of program: ${studyId} from RDPC ${url} after ${retryConfig.retries} attempts: ${err}`
+    );
+    throw err;
+  });
+};
+
+type StreamState = {
+  currentPage: number;
+};
+
+export const analysisStream = async function* (
+  studyId: string,
+  url: string,
+  config?: {
+    chunkSize?: number;
+    state?: StreamState;
+  }
+): AsyncGenerator<Analysis[]> {
+  const chunkSize = config?.chunkSize || 1000;
+  const streamState: StreamState = {
+    currentPage: config?.state?.currentPage || 0,
+  };
+  while (true) {
+    const page = await fetchSeqExpAnalyses(
+      studyId,
+      url,
+      streamState.currentPage,
+      chunkSize
+    );
+    streamState.currentPage = streamState.currentPage + chunkSize;
+
+    if (page && page.length > 0) {
+      yield page;
+    } else {
+      break;
+    }
+  }
+};
+
+/**
+ * Extracts donor-run relation from analysis, this is done by the following steps:
+ * Creates inputAnalyses-run map for each donor by grouping runs by inputAnalyses,
+ * creates donor-inputAnalyses-run map by determining the latest run for each inputAnalyses,
+ * aggregates donor-inputAnalyses-run map by donor id.
+ * @param analyses RDPC analysis array
+ */
+export const toDonorCentric = (analyses: Analysis[]): DonorDocMap => {
+  const result = analyses.reduce<DonorDocMap>((acc, analysis) => {
+    const donorWithLatestRun = analysis.donors.reduce<DonorDocMap>(
+      (_acc, donor) => {
+        const inputAnalysesMap = _(analysis.runs)
+          .groupBy((run) => {
+            return (
+              HashCode(run.inputAnalyses[0].analysisId) +
+              HashCode(run.inputAnalyses[1]?.analysisId)
+            );
+          })
```

Comment:
Please correct me if I'm wrong...

There'll be runs with more than 2 input analyses, so this is probably better generalized as:
```
.groupBy(
  run => _(run.inputAnalyses)
    .map(a => a.analysisId)
    .map(HashCode)
    .orderBy()
    .join("_")
)
```
it's also a bit dangerous to do `HashCode(a) + HashCode(b)` because `1 + 3` and `2 + 2` both give you `4` so you get a key collision, hence the `.join("_")` which gives you a unique string key. 
Also I included the `.orderBy()` assuming orders are not significant, but please adjust accordingly (should `1_2` and `2_1` be considered .

I actually don't think the HashCode is necessary, it's actually safer to go without it because it's ambiguous whether there's a possibility for key collision.
If the intention was to produce a consistent size key for memory efficiency, it'd be better to wrap the join result in `HashCode` like so rather than each element in inputAnalyses:
```
.groupBy(
  run => HashCode(
     _(run.inputAnalyses)
      .map(a => a.analysisId)
      .orderBy()
      .join("_")
  )
)
```